chainer/testing/test.py

```diff
@@ -0,0 +1,180 @@
+import contextlib
```

Comment:
Isn't there any better naming than `test`?
Or how about just merging `function.py` and `link.py`?

Reply:
Do you have any suggestions? We could merge them together but would still need a new name.